docs/source/customizations/overview.mdx

```diff
@@ -117,4 +95,31 @@ class client,subgraph1,subgraph2 secondary;
 
 ---
 
-Next, see the documentation for your preferred [customization type](#customization-types).
+### Router service
+
+The router service is called right after the HTTP server. The `RouterRequest` contains HTTP headers and the body as a stream of byte arrays. The `RouterResponse` contains HTTP headers and the body as a stream of byte arrays. The router service handles Automatic Persisted Queries, parses the GraphQL request from JSON, calls the supergraph service, and serializes the GraphQL responses to JSON.
+
+### Supergraph service
+
+The supergraph service works on a `SupergraphRequest` containing HTTP headers and a GraphQL request object. The `SupergraphResponse` contains headers and a stream of GraphQL responses. That stream only contains one element for most queries—it can contain more if the query uses the `@defer` directive or subscriptions.
+
+The supergraph service calls into the query planner, which returns a query plan. The supergraph service then calls the execution service.
+
+### Execution service
+
+The execution service is tasked with executing the query plan. The `ExecutionRequest` contains the `SupergraphRequest` and the query plan. The `ExecutionResponse` has the same content as the `SupergraphResponse`.
+
+For each fetch node of the query plan, the execution service creates a subgraph request and then calls the subgraph plugins and service. Once the execution service has received all subgraph responses, it formats the GraphQL responses—removing unneeded data and propagating nulls—before sending it back to the supergraph plugin.
```

Comment:
Does the execution service also compose the subgraph responses into one response? Or does it send them separately?

```suggestion
For each fetch node of the query plan, the execution service creates a subgraph request and then calls the subgraph plugins and service. Once the execution service has received all subgraph responses, it formats the GraphQL responses—removing unneeded data and propagating nulls—before sending them back to the supergraph plugin.
```